src/mag_panel.py

```diff
@@ -115,7 +115,7 @@ def update_data(self, output):
 
 
         if self.GetPlotParam('mag_plot_type') == 2: # Set f to deltaB_perp/B0
-            if ~np.isnan(self.parent.btheta):
+            if not np.isnan(self.parent.btheta):
```

Comment:
technically `np.isnan` could return an ndarray, but it doesn't here... I don't know if am just paranoid around types but i think
```suggestion
            if not math.isnan(self.parent.btheta):
```
is better but you need to import that `math` module at the top

Reply:
numpy handles scalar values just fine. I try not to mix in `math` when I can avoid it since it can lead to some weirdness sometimes with numpy scalar types.  I'd rather keep all my arithmetic in the numpy universe than moving between numpy and the standard library

Reply:
that's fair. in this case it hopefully will fail as a runtime error if an ndarray is passed away (80% sure)